src/pages/MyPage.tsx

```diff
@@ -1,3 +1,128 @@
-export default function MyPage(){
-    return <text>마이 페이지</text>
+import { useState, useEffect, useCallback } from "react";
+import { getTierImageUrl } from "../constants/tierMap";
+// 유틸리티 함수
+import { formatDateToISO, getYearMonth } from "../utils/dateUtils";
+
+//컴포넌트 import 
+import HeatMapCalendar from "../components/mypage/HeatMapCalendar";
+import DailySummary from "../components/mypage/DailySummary";
+import ProblemListModal from "../components/mypage/ProblemListModal";
+
+// 타입 및 Mock 함수
+import { MyPageResponse } from "../types/mypage";
+import { getMockDashboard } from "../mock/mypage/myPageData";
+
+export default function MyPage() {
+  // 상태 관리
+  //data: 프로필, 잔디, 선택한 상세정보가 포함된 전체 데이터 
+  const [data, setData] = useState<MyPageResponse["response"] | null>(null); 
+  const [selectedDate, setSelectedDate] = useState<Date>(new Date());
+  const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
+
+  // 데이터 로드 함수 (useCallback으로 메모이제이션)
+  const loadData =  useCallback(async (date: Date, dateStr?: string): Promise<void> => {
+    const { year, month } = getYearMonth(date);
+    const result = getMockDashboard(year, month, dateStr);
+    setData(result.response);
+  }, []);
+
+  // 초기 진입 
+  useEffect(() => {
+    const today = new Date();
+    loadData(today); 
+  }, [loadData]);
+
+  // 날짜 클릭 핸들러 
+  const handleDateClick = useCallback((clickedDate: Date): void => {
+    setSelectedDate(clickedDate);
+    const dateStr = formatDateToISO(clickedDate);
+    loadData(clickedDate, dateStr);
+  }, [loadData]);
+
+  // 월 변경 핸들러
+  const handleMonthChange = useCallback((newDate: Date): void => {
+    loadData(newDate);
+  }, [loadData]);
+
+  // 모달 핸들러
+  const handleOpenModal = useCallback((): void => {
+    setIsModalOpen(true);
+  }, []);
+
+  const handleCloseModal = useCallback((): void => {
+    setIsModalOpen(false);
+  }, []);
+
+   // 로딩 중 UI
+  if (!data) {
+    return <div className="flex justify-center p-20 text-gray-500">Loading...</div>;
+  }
```

Comment:
```suggestion
//공통 api 훅 예시
const useFetchData = (url : string) =>{
   const [isLoading, setIsLoading] = useState<Boolean>(false);
   const [error,setError] = useState(null);
   const [data,setData] = useState(null);
   
   useEffect(()=> {
     const getData = async ( )=>{
         setIsLoad(true);;
         setError(null);
         
         //api 통신 로직
         
         //성공시 
         setData(response.data);
         
         //실패시
         setError(에러 메시지);
         
         //finally(공통)
         setIsLoading(false);
       };
       
       getData();
    },[url]);
       
```
`data`가 존재하지 않을 때 로딩 컴포넌트를 띄우는 방식 보다는 나중에 공통 api 훅 개발에서 위와 같이 로딩 상태를 관리하는 것이 좋은 것 같습니다.